neuralpp/symbolic/constants.py

```diff
@@ -16,20 +23,18 @@ def if_then_else_function(type_: Type) -> Expression:
 real_if_then_else_function = if_then_else_function(fractions.Fraction)
 
 
-def if_then_else(if_: Any, then_: Any, else_: Any) -> Expression:
-    if not type(then_) == type(else_):
-        raise TypeError(f"Expect then-clause ({type(then_)}) and else-clause ({type(else_)}) have the same type.")
-    match then_:
-        case bool():
-            return bool_if_then_else_function(if_, then_, else_)
-        case int():
-            return int_if_then_else_function(if_, then_, else_)
-        case float():
-            return float_if_then_else_function(if_, then_, else_)
-        case fractions.Fraction():
-            return real_if_then_else_function(if_, then_, else_)
-        case _:
-            raise TypeError(f"Unrecognized type {type(then_)}.")
+def if_then_else(if_: Any, then_: Any, else_: Any):
```

Comment:
Also redesigned `if_then_else` so it can take not only python objects but also our `Expression`s.